### PR TITLE
feat(EMS-3877-3879): business - turnover currency - save and back

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR adds "save and back" functionality to the "Business - Turnover currency" form.

## Resolution :heavy_check_mark:
- Create new POST route and controller.
- Update the GET controller's `pageVariables` to have a `SAVE_AND_BACK_URL`.
- Note: E2E test coverage already exists via the DRY `assertCurrencyFormFields` assertions. Instead, a `hasSaveAndBack: false;` flag has been removed.

## Miscellaneous :heavy_plus_sign:
- Remove `hasSaveAndBack` flag from the `assertCurrencyFormFields` assertions.
- Fix some `checkValue` vs `checkTextareaValue` related E2E tests.

### CODE REVIEW

File: e2e-tests/commands/shared-commands/assertions/assert-empty-contract-completion-date-field-values.js [hunk 1]
```diff
@@ -12,9 +12,9 @@ const {
  * Assert all CONTRACT_COMPLETION_DATE field values are empty.
  */
 const assertEmptyContractCompletionDateFieldValues = () => {
-  cy.checkValue(field(CONTRACT_COMPLETION_DATE).dayInput(), '');
-  cy.checkValue(field(CONTRACT_COMPLETION_DATE).monthInput(), '');
-  cy.checkValue(field(CONTRACT_COMPLETION_DATE).yearInput(), '');
+  field(CONTRACT_COMPLETION_DATE).dayInput().should('have.value', '');
```
Comment (Zainzzkk) [inline e2e-tests/commands/shared-commands/assertions/assert-empty-contract-completion-date-field-values.js:15]: the checkValue command?

File: e2e-tests/commands/shared-commands/assertions/assert-empty-requested-start-date-field-values.js [hunk 2]
```diff
@@ -10,9 +10,9 @@ const {
  * Assert all REQUESTED_START_DATE field values are empty.
  */
 const assertEmptyRequestedStartDateFieldValues = () => {
-  cy.checkValue(field(REQUESTED_START_DATE).dayInput(), '');
-  cy.checkValue(field(REQUESTED_START_DATE).monthInput(), '');
-  cy.checkValue(field(REQUESTED_START_DATE).yearInput(), '');
+  field(REQUESTED_START_DATE).dayInput().should('have.value', '');
```
Comment (Zainzzkk) [inline e2e-tests/commands/shared-commands/assertions/assert-empty-requested-start-date-field-values.js:13]: same as above

File: e2e-tests/commands/shared-commands/assertions/assert-empty-contract-completion-date-field-values.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/commands/shared-commands/assertions/assert-empty-contract-completion-date-field-values.js:15]: It does not work for date fields. It relies on an `.input` method, which is irrelevant for date fields (`dayInput` etc)

Need to create a new command for this.

I am not sure how this change got merged with failing tests, seems GHA did not correctly re-run some tests or something in [this PR](https://github.com/UK-Export-Finance/exip/pull/3141)

File: e2e-tests/commands/shared-commands/assertions/assert-empty-requested-start-date-field-values.js (hunk 2 above)
Comment (ttbarnes) [reply e2e-tests/commands/shared-commands/assertions/assert-empty-requested-start-date-field-values.js:13]: Same reply as above

File: e2e-tests/commands/shared-commands/assertions/assert-empty-contract-completion-date-field-values.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/commands/shared-commands/assertions/assert-empty-contract-completion-date-field-values.js:15]: Have created EMS-3898 to address this.

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**